### DISCUSSION TITLE
Cant run ast-grep 0.7.2 in github actions shell

### DISCUSSION THREAD

**Discussion by 93578237** [ast-grep/ast-grep#500] [Q&A] [answered]
```yml
- run: sg scan --color always
```
This step fails in 0.7.2 (0.6.6 is working) with this error https://github.com/ast-grep/ast-grep/commit/66cbc50c7844f6b6b2892c93a2c1a5d8a82d3a14. What is wrong?

```
Run sg scan --color always
  sg scan --color always
  shell: /usr/bin/bash -e {0}
Error: Only one rule can scan code from StdIn.
Help: Please use `--rule path/to/rule.yml` to choose the rule.
See also: https://ast-grep.github.io/guide/tooling-overview.html#parse-code-from-stdin
```

**Comment by 93578237** [comment]
GHA do not create TTY for their runners https://github.com/actions/runner/issues/241

**Comment by HerringtonDarkholme** [comment, marked as answer]
~~This is because `sg scan` is invoked from another process.~~ Oh, GHA does not create TTY.

You can use `--no-stdin` to disable this behavior.

https://ast-grep.github.io/guide/tooling-overview.html#disable-stdin-mode

**Reply by HerringtonDarkholme** [reply]
hmmm, looks like stdin mode is very confusing. I think it is better for users to opt-in by an option instead of relying on TTY.

**Reply by 93578237** [reply]
thx!

**Reply by HerringtonDarkholme** [reply]
Would you mind use it later with 0.8.0. I'm going to change the `--no-stdin` to `--stdin` for better usability.

**Reply by 93578237** [reply]
No problem, I would remove it for 8.0.0
